### PR TITLE
Support ability to show more than one metric on a single graph.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**
I've refactored quite a bit of the code on the server side to facilitate this and make it a little cleaner.

### CODE REVIEW

**Comment by LATaylor-guardian** [comment]
FYI In light of this PR, if we show two metrics on a single graph, it's currently unclear which line equates to which metric on the graph (if its a linegraph). I will try to incorporate the following Chartist plugin: https://github.com/CodeYellowBV/chartist-plugin-legend to address this in a seperate PR.

File: app/com/gu/bard/metrics/PageInsightGraphs.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Provides functions for the computation of Insight metrics in graph form for rendering.
+ */
+class PageInsightGraphs(graphSettingsMap: Map[String, GraphSettings], pageInsights: Seq[Insight], weekRanges: Seq[WeekRange]) extends StrictLogging with Graphing {
+
+  def totalPostLikeReactions: Option[Graph] = {
+    val GraphSettingsKey = "totalPostLikeReactions"
+    getGraph(GraphSettingsKey)
+  }
```
Comment (oilnam) [inline app/com/gu/bard/metrics/PageInsightGraphs.scala]: Shouldn't `GraphSettingsKey` be lowercase? 

Actually can we just do `getGraph("totalPostLikeReactions")`?

Comment (LATaylor-guardian) [inline app/com/gu/bard/metrics/PageInsightGraphs.scala]: Constants in Scala are denoted with an uppercased variable name. It's in it's own variable as I plan to seperate them out as constants to be shared by `PageInsightsPageSettings` when I get round to it.

**Comment by oilnam** [comment]
- I like how you reduced the boilerplate code a lot for adding new graphs :)
- `PageService` is nice and easy to follow
- the `getGraph` function on the other hand is pretty complicated. I'd propose some refactoring but looks like you were planning on doing it already?

👍 to me!

File: app/com/gu/bard/metrics/PageInsightGraphs.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      val insights = pageInsights.find(_.getName == metricName).map(_.getValues.asScala.toList).toList.flatten
+      val graphData = GraphData(values = graphDataFn(insights))
+
+      Metric(metricSettings, graphData)
+    }
+
+    val maybeGraph = graphSettingsMap.get(graphSettingsKey) map { graphSettings =>
+      val metricNames = graphSettings.metricSettings.map(_.fbMetricName)
+
+      val metrics = metricNames flatMap { metricName =>
```
Comment (oilnam) [inline app/com/gu/bard/metrics/PageInsightGraphs.scala]: Maybe we could single this out in a function? It's not too obvious what this bit of code is doing and maybe adding some semantics would help (me at least)

Comment (LATaylor-guardian) [inline app/com/gu/bard/metrics/PageInsightGraphs.scala]: Yeah, I agree getGraph is a bit complicated. I can definitely make it more readable at the expensive of brevity. Will update the PR with something.

**Comment by LATaylor-guardian** [comment]
Merging.

**Pull request merged**